### ISSUE TITLE
Config system for render template aliasing

### ISSUE THREAD

**Issue by mattrandallbecker** [abcxyz/abc#46]
### TL;DR

For better demos, first time adoption, and user customization allow aliased templates.

Ideally this is a local config file that is checked by the CLI, there will be some section that is similar to:

abc.config
{
  "aliases": [
    ["hello", "github.com/abcxyz/abc.git//examples/templates/render/hello_jupiter"],
    ["go", "github.com/mattrandallbecker/my_cool_go_starter_template"],
  ]
}

When you run render the cli would read that config and create a map of all the aliases, and if the template argument doesn't look like a path, check aliases.

This would benefit our documentation by being able to change from:
```
abc templates render \
  github.com/abcxyz/abc.git//examples/templates/render/hello_jupiter
```
to
```
abc templates render hello
```

but more importantly allow me to create my aliases for common service templates without adding them to my zsh file.

**Comment by pdewilde** [comment]
Maybe TOML would be a good choice for config language: https://toml.io/en/

(I'm always looking for an excuse to stan TOML)

**Comment by sethvargo** [comment]
I'm generally against these kinds of mappings, because it makes commands less portable (and more difficult to debug)

**Comment by pdewilde** [comment]
What about adding as part of the release? Or having the CLI tool poll some alias file on github so we can have a similar experience to docker or something with friendly names

**Comment by sethvargo** [comment]
Dave and I talked about this extensively. Every ecosystem that has tried to do "top-level namespacing" has inevitably had to walk it back, including Docker

**Comment by NishThakkar** [comment]
I see the value of namespacing, but am surprised to see "github.com/abcxyz/abc.git" is required in all cases. Given the scale we're designing for, do we expect other repos and non-github.com?

**Comment by NishThakkar** [comment]
okay, I just saw the example that has Matt's repo, wouldn't we want a connecting in abcxyz for that so that it's clear it's vetted?
